lxd/storage_pools.go

```diff
@@ -557,6 +558,57 @@ func storagePoolDelete(d *Daemon, r *http.Request) Response {
 		return EmptySyncResponse
 	}
 
+	volumeNames, err := d.cluster.StoragePoolVolumesGetNames(poolID)
+	if err != nil {
+		return InternalError(err)
+	}
+
+	for _, volume := range volumeNames {
+		pools, err := d.cluster.ImageGetPools(volume)
+		if err != nil {
+			return InternalError(err)
+		}
+
+		if len(pools) == 1 {
+			imgID, imgInfo, err := d.cluster.ImageGet(volume, false, false)
+			if err != nil {
+				return InternalError(err)
+			}
+
+			err = doDeleteImageFromPool(d.State(), imgInfo.Fingerprint, poolName)
+			if err != nil {
+				return InternalError(err)
+			}
+
+			// Remove main image file.
+			fname := shared.VarPath("images", imgInfo.Fingerprint)
+			if shared.PathExists(fname) {
+				err = os.Remove(fname)
+				if err != nil {
+					logger.Debugf("Error deleting image file %s: %s", fname, err)
+				}
+			}
+
+			// Remove the rootfs file for the image.
+			fname = shared.VarPath("images", imgInfo.Fingerprint) + ".rootfs"
+			if shared.PathExists(fname) {
+				err = os.Remove(fname)
+				if err != nil {
+					logger.Debugf("Error deleting image file %s: %s", fname, err)
+				}
+			}
+
+			// Remove the database entry for the image.
+			d.cluster.ImageDelete(imgID)
```

Comment:
This is pretty wrong here and the reason why the CEPH test fails.

We need to only remove the image from the pool we're deleting, we very much MUST keep the image in the database and in the "images" directory.